tonic/src/codec/compression.rs

```diff
@@ -104,13 +148,28 @@ impl CompressionEncoding {
         let header_value = map.get(ACCEPT_ENCODING_HEADER)?;
         let header_value_str = header_value.to_str().ok()?;
 
-        split_by_comma(header_value_str).find_map(|value| match value {
-            #[cfg(feature = "gzip")]
-            "gzip" => Some(CompressionEncoding::Gzip),
-            #[cfg(feature = "zstd")]
-            "zstd" => Some(CompressionEncoding::Zstd),
-            _ => None,
-        })
+        // Get the highest priority supported encoding
+        split_by_comma(header_value_str)
```

Comment:
Maybe some unit tests would be nice, e.g. to test that this works in the case where only one of them is enabled, when both, etc.